src/test/java/com/getindata/connectors/http/internal/sink/httpclient/JavaNetSinkHttpClientConnectionTest.java

```diff
@@ -62,6 +66,33 @@ public void testHttpConnection() {
             batchRequestSubmitterFactory);
     }
 
+    @Test
+    public void testHttpPostRequestCallbackWithException() {
+        wireMockServer = new WireMockServer(SERVER_PORT);
+        wireMockServer.start();
+        mockEndPoint(wireMockServer);
+
+        try {
+            JavaNetSinkHttpClient client =
+                    new JavaNetSinkHttpClient(
+                            properties,
+                            new TestPostRequestCallbackWithException(),
+                            headerPreprocessor,
+                            perRequestSubmitterFactory);
+            HttpSinkRequestEntry requestEntry = new HttpSinkRequestEntry("GET", new byte[0]);
+            SinkHttpClientResponse response =
+                    client.putRequests(
+                            Collections.singletonList(requestEntry),
+                            "https://localhost:" + HTTPS_SERVER_PORT + ENDPOINT
+                    ).get();
+
+            assertThat(response.getSuccessfulRequests()).isEmpty();
+            assertThat(response.getFailedRequests()).isNotEmpty();
+        } catch (Exception e) {
```

Comment:
do we need the catch? Why not put the checked exceptions on the message then the stack trace will point to the real error.